mouseion/scripts/s2_train_and_build_index.py

```diff
@@ -0,0 +1,77 @@
+###
+#
+# Copyright Computelify, Inc.
+#
+# This trains and builds an index.
+#
+# Author: Steven Dake <steve@computelify.com>
+# License: ASL2.
+
+import subprocess
+import pathlib
+
+def get_base_path() -> pathlib.Path:
+    """
+    Get base path.
+    """
+    return pathlib.Path(__file__).parent.parent.resolve()
+
+def get_mouseion_source_path() -> pathlib.Path:
+    """
+    Get Mousieon source path.
+    """
+    return get_base_path / "mouseon"
+
+def build_workload(base_path: pathlib.Path) -> list:
+    """
+    Constructs the command as a list of arguments to be executed.
+
+    Args:
+        base_path (pathlib.Path): The base path of the project.
+
+    Returns:
+        list: The command as a list of arguments.
+    """
+    datacollection_specs_path = base_path / "datacollection_specs"
+    realnews_path = base_path / "knowledge.d/realnews.d"
+    mouseion_source_path = base_path / "mouseion"
+
+    # The hyperfine bench tool requires a single string command.
+    workload_cmd = (
+        f"python '{mouseion_source_path / 'train_and_build_index.py'}' "
+        f"--spec '{datacollection_specs_path / 'train_realnews_dataset_spec.json'}' "
+        f"--index-type IVF131072,PQ32 "
+        f"--output '{realnews_path / 'realnews.mouseion'}' "
+        f"--use-gpus "
+        f"--batch-size 32768"
```

Comment:
f-strings aren't wrong here, but static analysis tools like `pyflakes` will complain about them; the tools assume that we forgot to interpolate something or made a syntax mistake.

Reply:
objections to the use of the black static analyzer?
https://black.readthedocs.io/en/stable/